src/material-experimental/mdc-slider/slider.ts

```diff
@@ -221,89 +316,7 @@ export class MatSlider implements AfterViewInit, OnChanges, OnDestroy, ControlVa
-          .addEventListener(evtType, handler, passiveListenerOptions);
-      });
-    },
-    deregisterThumbContainerInteractionHandler: (evtType, handler) => {
-      this._thumbContainer.nativeElement
-        .removeEventListener(evtType, handler, passiveListenerOptions);
-    },
-    registerBodyInteractionHandler: (evtType, handler) =>
-        // Body event handlers (which handle thumb sliding) cannot be passive as they will
-        // prevent the default behavior. Additionally we can't run these event handlers
-        // outside of the Angular zone because we rely on the events to cause the component
-        // tree to be re-checked.
-        document.body.addEventListener(evtType, handler),
-    deregisterBodyInteractionHandler: (evtType, handler) =>
-        document.body.removeEventListener(evtType, handler),
-    registerResizeHandler: (handler) => {
-      // The resize handler is currently responsible for detecting slider dimension
-      // changes and therefore doesn't cause a value change that needs to be propagated.
-      this._ngZone.runOutsideAngular(() => window.addEventListener('resize', handler));
-    },
-    deregisterResizeHandler: (handler) => window.removeEventListener('resize', handler),
-    notifyInput:
-        () => {
-          const newValue = this._foundation.getValue();
-          // MDC currently fires the input event multiple times.
-          // TODO(devversion): remove this check once the input notifications are fixed.
-          if (newValue !== this.value) {
-            this.value = newValue;
-            this.input.emit(this._createChangeEvent(newValue));
-          }
-        },
-    notifyChange:
-        () => {
-          // TODO(devversion): bug in MDC where only the "change" event is emitted if a keypress
-          // updated the value. Material and native range sliders also emit an input event.
-          // Usually we sync the "value" in the "input" event, but as a workaround we now sync
-          // the value in the "change" event.
-          this.value = this._foundation.getValue();
-          this._emitChangeEvent(this.value!);
-        },
-    setThumbContainerStyleProperty:
-        (propertyName, value) => {
-          this._thumbContainer.nativeElement.style.setProperty(propertyName, value);
-        },
-    setTrackStyleProperty:
-        (propertyName, value) => {
-          this._track.nativeElement.style.setProperty(propertyName, value);
-        },
-    setMarkerValue:
-        () => {
-          // Mark the component for check as the thumb label needs to be re-rendered.
-          this._changeDetectorRef.markForCheck();
-        },
-    setTrackMarkers:
-        (step, max, min) => {
-          this._trackMarker.nativeElement.style.setProperty(
-              'background', this._getTrackMarkersBackground(min, max, step));
-        },
-    isRTL: () => this._isRtl(),
-  };
+  private _sliderAdapter: MDCSliderAdapter = new SliderAdapter(this);
```

Comment:
You don't need to keep this reference anymore. Just do new MDCSliderFoundation(new SliderAdapter(this)); below

Reply:
Got it.